lib/analyzer.js

```diff
@@ -39,6 +39,20 @@ var EMPTY_METADATA = {elements: [], features: [], behaviors: []};
  */
 
 /**
+ * espree's representation of a parsed html document
+ * @typedef {Object} JSAST
```

Comment:
How about, instead of using `{Object}` here, something like the below?

```js
/**
 * An AST Node as defined by
 * [SpiderMonkey's reflect API](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Parser_API#Node_objects)
 * and as produced by esmangle.
 * @constructor
 */
function Node() {}
/** @type {string} */
Node.prototype.type;
/** @type {?SourceLocation} */
Node.prototype.loc;

/**
 * Also defined by SpiderMonkey's reflect API.
 * @constructor
 */
function SourceLocation() {}
/** @type {string | null} */
SourceLocation.prototype.source;
/** @type {Position | null} */
SourceLocation.prototype.start;
/** @type {Position | null} */
SourceLocation.prototype.end;

/**
 * Also defined by SpiderMonkey's reflect API.
 * @constructor
 */
function Position() {}
/** one-indexed  @type {number} */
Position.prototype.line;
/** zero-indexed @type {number} */
Position.prototype.column;
```